earth_enterprise/src/fusion/autoingest/plugins.src/RasterProject.src

```diff
@@ -1057,19 +1054,14 @@ uint32 RasterProjectAssetVersionImplD::BuildIndex(
            i+1, genInfo.size(), timeDiff.count());
     prevTime = curTime;
 
-    if (type == AssetDefs::Imagery) {
-      FindNeededImageryInsets(genInfo[i].coverage,
-                              insetInfoPtrs, i,
-                              neededIndexes,
-                              beginMinifyLevel,
-                              endMinifyLevel);
-    } else {
-      FindNeededTerrainInsets(genInfo[i].coverage,
-                              insetInfoPtrs, i,
-                              neededIndexes,
-                              beginMinifyLevel,
-                              endMinifyLevel);
-    }
+    overlapEnvelope<RasterProductAssetVersion> env (
+        type, genInfo[i].coverage, insetInfoPtrs, i,
+        beginMinifyLevel, endMinifyLevel, i);
+
+    std::shared_ptr<OverlapCalculator<RasterProductAssetVersion>> calc =
+        std::make_shared<OverlapCalculator<RasterProductAssetVersion>>(env);
+    calc->PreprocessForInset();
+    neededIndexes = std::move(calc->GetOverlapForLevel());
     neededIndexes.push_back(i); // each inset always intersects itself
```

Comment:
I think there may be a problem here. Since we're adding the inset outside of the calc object the version of neededIndexes that's inside the object will not have the extra element.

Reply:
fixed